### PR TITLE
Bbonev/disco fused quadrature

### PR DESCRIPTION
**Pull request by bonevbs**
Fused the multiplication of the quadrature weights into Psi, to slightly reduce the memory footprint of the DISCO convolution

### CODE REVIEW

File: torch_harmonics/convolution.py [hunk 1]
```diff
@@ -167,7 +167,11 @@ def _normalize_convolution_tensor_s2(psi_idx, psi_vals, in_shape, out_shape, ker
                 iidx = torch.argwhere((idx[0] == ik) & (idx[2] == ilat))
                 # normalize, while summing also over the input longitude dimension here as this is not available for the output
                 vnorm = torch.sum(psi_vals[iidx] * q[iidx])
-                psi_vals[iidx] = psi_vals[iidx] / (vnorm + eps)
+                if merge_quadrature:
+                    # the correction factor accounts for the difference in longitudinal grid points when the input vector is upscaled
+                    psi_vals[iidx] = psi_vals[iidx] * q[iidx] * nlon_in / nlon_out / (vnorm + eps)
```
Comment (azrael417) [inline torch_harmonics/convolution.py:172]: Why do we need the nlon_in / nlon_out ?

**Review by azrael417 — APPROVED** [review]
lgtm, but please check the comment

File: torch_harmonics/convolution.py (hunk 1 above)
Comment (bonevbs) [reply torch_harmonics/convolution.py:172]: the quadrature weights that are used to normalize the transpose use the output resolution, however quadrature needs to run over the input. This has to be accounted for using this factor.
